db/range_del_aggregator.cc

```diff
@@ -10,20 +10,40 @@
 
 namespace rocksdb {
 
+struct TruncatedRangeTombstone {
+  explicit TruncatedRangeTombstone(RangeTombstone t, bool truncated = false)
+      : tombstone(t), is_truncated(truncated) {}
+
+  RangeTombstone tombstone;
```

Comment:
There seem to be a lot of mechanical changes below that believe could be avoided if you used inheritance instead of embedding. That is: `struct TruncatedRangeTombstone : public RangeTombstone`. Is there a reason you went for embedding?

Reply:
No particular reason; I just did what first came to mind since I was thinking more about the collapsed case. It's arguably a bit cleaner this way to support `RangeDelIter::Tombstone`, but I agree that having to do `->tombstone` everywhere else isn't worth it. I'll change this to use inheritance.

Reply:
I retract my earlier comment about `RangeDelIter::Tombstone` being cleaner with my original way. Object slicing FTW for once.